src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,196 @@
+package duke.parser;
+
+import duke.core.Duke;
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+/**
+ * The `Parser` class handles the parsing of user commands in Duke.
+ * It interprets commands and converts them into corresponding Duke actions.
+ */
+public class Parser {
+
+    /**
+     * The command to be parsed.
+     */
+    private final String command;
+
+    /**
+     * Constructs a `Parser` with the given command.
+     *
+     * @param command The command to be parsed.
+     */
+    public Parser(String command) {
+        this.command = command;
+    }
+
+    /**
+     * Parses the "bye" command.
+     *
+     * @return True if the input is a "bye" command, false otherwise.
+     * @throws DukeException If there is an issue parsing the command.
+     */
+    public boolean parseBye() throws DukeException {
+        String[] parts = command.split(" ");
+        if (parts.length == 1 && parts[0].equalsIgnoreCase(Duke.Command.BYE.name())) {
+            return true;
+        } else {
+            throw new DukeException("Invalid format. Please use 'bye'.");
+        }
+    }
+
+    /**
+     * Parses the "list" command.
+     *
+     * @return True if the input is a "list" command, false otherwise.
+     * @throws DukeException If there is an issue parsing the command.
+     */
+    public boolean parseList() throws DukeException {
+        String[] parts = command.split(" ");
+        if (parts.length == 1 && parts[0].equalsIgnoreCase(Duke.Command.LIST.name())) {
+            return true;
+        } else {
+            throw new DukeException("Invalid format. Please use 'list'.");
+        }
+    }
+
+    /**
+     * Parses the "find" command to search for tasks with a keyword.
+     *
+     * @return The keyword to search for.
+     * @throws DukeException If the command format is invalid.
+     */
+    public String parseFind() throws DukeException {
+
+        String[] parts = command.split(" ", 2);
+        if (parts.length < 2 || parts[1].isEmpty()) {
+            throw new DukeException("Invalid format. Please use 'find <keyword>'.");
+        }
+        return parts[1].trim();
+    }
+
+    /**
+     * Parses the delete command and returns the index of the task to be deleted.
+     *
+     * @return The index of the task to be deleted.
+     * @throws DukeException If the command is in an invalid format.
+     */
+    public int parseDelete() throws DukeException {
+        String[] part = command.split(" ");
```

Comment:
Should this be renamed `parts`? I noticed a change in the variable naming for the next few methods as well.